### PR TITLE
Adding more options for upload disabling

### PR DESCRIPTION
**Pull request by Akatsuki-Ousawa**
Added a few more options for Upload disabling.

### CODE REVIEW

File: router/apiHandler.go [hunk 1]
```diff
@@ -173,15 +179,21 @@ func ApiUploadHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func ApiUpdateHandler(w http.ResponseWriter, r *http.Request) {
-	if config.UploadsDisabled {
+	user := model.User{}
```
Comment (sfan5) [inline router/apiHandler.go]: This is an empty struct, you need to actually read the user from the db

File: router/apiHandler.go [hunk 2]
```diff
@@ -96,13 +96,19 @@ func ApiViewHandler(w http.ResponseWriter, r *http.Request) {
 }
 
 func ApiUploadHandler(w http.ResponseWriter, r *http.Request) {
-	if config.UploadsDisabled {
+	user := model.User{}
```
Comment (sfan5) [inline router/apiHandler.go]: same applies here

File: router/apiHandler.go (hunk 1 above)
Comment (Akatsuki-Ousawa) [inline router/apiHandler.go]: But where is it read in the old file? I mean I just moved it up a few lines and from my understanding nothing redecleared it to fill it

Comment (sfan5) [inline router/apiHandler.go]: https://github.com/NyaaPantsu/nyaa/pull/615/files#diff-021a8ffd4664e7215a6a15d63ed60234R197

Comment (Akatsuki-Ousawa) [inline router/apiHandler.go]: changed it a bit pls take a look

File: router/apiHandler.go [hunk 3]
```diff
@@ -96,14 +96,20 @@ func ApiViewHandler(w http.ResponseWriter, r *http.Request) {
...
+	if config.UploadsDisabled && config.AdminsAreStillAllowedTo && user.Status != 2 && config.TrustedUsersAreStillAllowedTo && user.Status != 1 {
+		http.Error(w, "Error uploads are disabled", http.StatusBadRequest)
+		return
+	} else if config.UploadsDisabled && !config.AdminsAreStillAllowedTo && user.Status == 2 {
+		http.Error(w, "Error uploads are disabled", http.StatusBadRequest)
+		return
+	} else if config.UploadsDisabled && !config.TrustedUsersAreStillAllowedTo && user.Status == 1 {
+		http.Error(w, "Error uploads are disabled", http.StatusBadRequest)
+		return
+	}
```
Comment (sfan5) [inline router/apiHandler.go]: Hm, the config.UploadsDisabled should probably be moved to a seperate if-clause for better readability

Comment (Akatsuki-Ousawa) [inline router/apiHandler.go]: So more like an outer if clause for config.UploadsDisabled and an inner for the rest?

Comment (sfan5) [inline router/apiHandler.go]: Yep

Comment (Akatsuki-Ousawa) [inline router/apiHandler.go]: changed it

**Pull request merged**